bodo/io/iceberg/common.py

```diff
@@ -196,93 +172,30 @@ class IcebergConnectionType(types.Type):
     """
 
     def __init__(self, name):  # pragma: no cover
-        super().__init__(
-            name=name,
-        )
+        super().__init__(name=name)
 
     def get_conn_str(self) -> str:
         raise NotImplementedError("IcebergConnectionType should not be instantiated")
 
 
-def format_iceberg_conn(conn_str: str) -> str:
-    """
-    Determine if connection string points to an Iceberg database and reconstruct
-    the correct connection string needed to connect to the Iceberg metastore
-    """
-
-    parse_res = urlparse(conn_str)
-    if not conn_str.startswith("iceberg+glue") and parse_res.scheme not in (
-        "iceberg",
-        "iceberg+file",
-        "iceberg+s3",
-        "iceberg+thrift",
-        "iceberg+http",
-        "iceberg+https",
-        "iceberg+snowflake",
-        "iceberg+abfs",
-        "iceberg+abfss",
-        "iceberg+rest",
-        "iceberg+arn",
-    ):
-        raise BodoError(
-            "'con' must start with one of the following: 'iceberg://', 'iceberg+file://', "
-            "'iceberg+s3://', 'iceberg+thrift://', 'iceberg+http://', 'iceberg+https://', 'iceberg+glue', 'iceberg+snowflake://', "
-            "'iceberg+abfs://', 'iceberg+abfss://', 'iceberg+rest://', 'iceberg+arn'"
-        )
-
-    # Remove Iceberg Prefix when using Internally
-    conn_str = conn_str.removeprefix("iceberg+").removeprefix("iceberg://")
-
-    # Reformat Snowflake connection string to be iceberg-connector compatible
-    if conn_str.startswith("snowflake://"):
-        from bodo.io.snowflake import parse_conn_str
-
-        conn_contents = parse_conn_str(conn_str)
-        account: str = conn_contents.pop("account")
-        # Flatten Session Parameters
-        session_params = conn_contents.pop("session_parameters", {})
-        conn_contents.update(session_params)
-        # Remove Snowflake Specific Parameters
-        conn_contents.pop("warehouse", None)
-        conn_contents.pop("database", None)
-        conn_contents.pop("schema", None)
-        conn_str = (
-            f"snowflake://{account}.snowflakecomputing.com/?{urlencode(conn_contents)}"
-        )
+def _fs_from_file_path(file_path: str, io: FileIO) -> FileSystem:
```

Comment:
Mostly a duplicate of a internal function from PyIceberg, but slightly modified to take use of our monkey-patch